examples/nlp/cai/generate_rl_cai_dataset.py

```diff
@@ -0,0 +1,741 @@
+):
+    """
+    @param batch_size: inference batch size
+    @param temperatures: how many temperatures to use for generation per prompt
+    @param red_teaming_dataset_path: path to Anthropic red teaming prompt attempts.
+    @param port_num: inference service port number.
+    @return:
+    """
+    assert batch_size > 0
+    assert isinstance(temperatures, List) or isinstance(temperatures, int)
+    if isinstance(temperatures, int):
+        temperatures = [temperatures]
+
+    red_teaming_prompts = get_red_team_train_human_prompts(red_teaming_dataset_path)
+    # DEBUG DEBUG DEBUG - TODO REMOVE
+    # red_teaming_prompts = red_teaming_prompts[:1 * batch_size]
+
+    all_samples = []
+    samples_per_temperature = {}
+
+    for batch_index in tqdm(range(0, len(red_teaming_prompts), batch_size), desc="Batch #"):
+        red_teaming_prompts_list = red_teaming_prompts[batch_index : batch_index + batch_size]
+        if len(red_teaming_prompts_list) < batch_size:
+            break
+
+        for t in tqdm(temperatures, desc="Temperatures"):
+            samples = []
+
+            # call model
+            rlaif_batch_samples = generate_responses_batch(red_teaming_prompts_list, temperature=t, port_num=port_num)
+            samples.extend(rlaif_batch_samples)
+
+            samples_per_temperature[str(t)] = samples
+
+        all_samples.extend(join_responses(samples_per_temperature))
+
+    return all_samples
+
+
+def generate_responses_batch(prompt_list: list, temperature: int, port_num: int):
+    assert isinstance(prompt_list, list)
+    num_prompts = len(prompt_list)
+
+    # get initial response
+    prompts = [ChatPromptTemplate.apply_prompt_template(p) for p in prompt_list]
+
+    responses = model_remote_inference(prompts, temperature=temperature, port_num=port_num)
+    assert len(responses) == num_prompts
+    stripped_responses = [ChatPromptTemplate.extract_response(r) for r in responses]
+
+    s_batch = []
+    for i in range(num_prompts):
+        s = dict(prompt=prompt_list[i], response=stripped_responses[i])
+        s_batch.append(s)
+
+    return s_batch
+
+
+def model_remote_inference(prompt, port_num=5656, temperature=1.0):
+    if not isinstance(prompt, list):
+        prompt = [prompt]
+
+    headers = {"Content-Type": "application/json"}
+
+    def request_data(request):
+        resp = requests.put("http://localhost:{}/generate".format(port_num), data=json.dumps(request), headers=headers)
+        resp_json = resp.json()
+        resp_sentences = resp_json["sentences"]
+        return resp_sentences
+
+    data = {
+        "sentences": prompt,
+        "tokens_to_generate": 1024,
+        "temperature": temperature,
+        "add_BOS": True,
+        "top_k": 50,
+        "top_p": 0.95,
+        "greedy": False,
+        "all_probs": False,
+        "repetition_penalty": 1,
+        "min_tokens_to_generate": 1,
+        "end_strings": ["<extra_id_1>"],
+    }
```

Comment:
Can you please parametrize this json obj? If someone wants to try generating 2048 tokens, it would be helpful to not have to edit code. 

hydra may be too heavy for this script, so perhaps plumbing these via [add_argument_group](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_argument_group) would be a good middle ground.

Reply:
Thank you for your suggestion.
I've implemented the **add_argument_group** approach to parameterize the JSON object as you recommended.

Reply:
Thanks, done.
we have implemented the use of 'add_argument_group' to parameterize the JSON object as you recommended.